server.js

```diff
@@ -753,6 +753,18 @@ function exit() {
 
 process.on('SIGINT', exit);
```

Comment:
This is an existing handler for ctrl-c, it's probably better to see what this function does and modify it rather than create subtly different keypress handler.

Reply:
I would have used this if it would work. Tried every option. Try it out. I have implemented the restart when you click on your network adapter in the web interface.
Once you have done this, this exit method stops working well.
The method I implemented is the one reliable.

Reply:
I think this is what happens:
ctrl + C is still associated with the parent process, however that is eliminated and the link to the child is removed. When you call it, you get back to the command line, but the new running process is not eliminated. It still writes to the terminal and its process is still active. If you want to restart the server, you get error messages. 

However, the command line is inherited from the original process and as such you can listen to its input. Since there are no other command line inputs for the server, it seems a feasible solution.

Reply:
Yeah, this approach gets the job done at the end of the day.

One behavior that should be noted is that if someone closes their terminal instead of pressing ctrl-c it won't be able to terminate the detached process

In general this approach is hacky enough that it would be nice to get comments on each of the functions describing what they're doing

Reply:
Ouch.. I was not testing that. Thats a side effect of not catching SIGINT I guess.
I think this needs more work. We have to find a way how the normal SIGINT behavior works.

Reply:
Maybe we can create a start file that is called index.js and server is really just a child process of index.js. index.js is never killed and as such can handle a rightful exit.

Reply:
Once upon a time I did create an index.js file that runs server.js as a child process. You can try it out and see if it accomplishes what you're looking for

Reply:
See #105, I took the idea of the forever-based index.js and added support for a 'restart' message

Reply:
perfect!